lib/src/rules/unnecessary_parenthesis.dart

```diff
@@ -121,6 +133,18 @@ class _Visitor extends SimpleAstVisitor<void> {
       return;
     }
 
+    // `foo = (a == b)` is OK, `return (count != 0)` is OK.
+    if (expression is BinaryExpression &&
+        (expression.operator.type == TokenType.EQ_EQ ||
+            expression.operator.type == TokenType.BANG_EQ)) {
+      if (parent is AssignmentExpression ||
+          parent is VariableDeclaration ||
+          parent is ReturnStatement ||
+          parent is YieldStatement) {
```

Comment:
One of the examples in the issue is in the context of a constructor initializer, but this doesn't cover that case (the parent will be a `ConstructorFieldInitializer`.

Reply:
@oprypin: could you add this case (and a corresponding test)?

Reply:
Added ConstructorFieldInitializer (+test)